data_linter/validators/pandas_validator.py

```diff
@@ -362,100 +364,43 @@ def _parse_data_to_pandas(filepath: str, table_params: dict, metadata: dict):
-
     # Set the reader type
     if filepath.startswith("s3://"):
-        reader_fs = fs.S3FileSystem(region="eu-west-1")
-        fp_for_file_reader = filepath.replace("s3://", "", 1)
-
+        reader = wr.s3
     else:
-        reader_fs = fs.LocalFileSystem()
-        fp_for_file_reader = filepath
-
-    with reader_fs.open_input_stream(fp_for_file_reader) as f:
-        if "csv" in metadata["file_format"]:
-
-            # Safer CSV load for newlines_in_values set to True
-            if table_params.get("expect-header", True):
-                po = csv.ParseOptions(newlines_in_values=True)
-            else:
-                po = csv.ParseOptions(
-                    newlines_in_values=True, column_names=meta_col_names
-                )
-
-            if ts_as_str_schema:
-                co = csv.ConvertOptions(column_types=ts_as_str_schema)
-            else:
-                co = None
-
-            df = pa_read_csv_to_pandas(
-                input_file=f,
-                schema=arrow_schema,
-                expect_full_schema=False,
-                parse_options=po,
-                convert_options=co,
-            )
-            # dates/datetimes == string
-
-        elif "json" in metadata["file_format"]:
-
-            po = json.ParseOptions(
-                newlines_in_values=True,
-                explicit_schema=ts_as_str_schema if ts_as_str_schema else None,
-            )
-
-            df = pa_read_json_to_pandas(
-                input_file=f,
-                schema=arrow_schema,
-                expect_full_schema=False,
-                parse_options=po,
-            )
-            # dates/datetimes == string
+        reader = pd
+
+    # read the data
+    if "csv" in metadata["file_format"]:
+        df = reader.read_csv(filepath, dtype=str, low_memory=False)
+    elif "json" in metadata["file_format"]:
+        df = reader.read_json(filepath, dtype=str, lines=True)
+    elif "parquet" in metadata["file_format"]:
+        df = reader.read_parquet(filepath)
+    else:
+        raise ValueError(f"Unknown file_format in metadata: {metadata['file_format']}.")
 
-        elif "parquet" in metadata["file_format"]:
-            df = arrow_to_pandas(pq.read_table(f))
-            # dates/datetimes == datetime / date
+    # eliminate case sensitivity, if requested
+    if table_params.get("headers-ignore-case"):
+        for c in metadata["columns"]:
+            c["name"] = c["name"].lower()
+        df.columns = [c.lower() for c in df.columns]
 
-        else:
-            raise ValueError(
-                f"Unknown file_format in metadata: {metadata['file_format']}."
-            )
+    # cast table column by column, except timestamps
+    for c in metadata["columns"]:
+        if not c["type_category"].startswith("timestamp"):
+            df[c["name"]] = cast_pandas_column_to_schema(df[c["name"]], metacol=c)
+    # df = cast_pandas_table_to_schema(df, metadata)
```

Comment:
This conversion only needs to happen for CSV or JSON. For parquet we should assume metadata (we should probably test this in a later release). But this tool is more for rogue CSV / JSONL files anyway, so lets focus on this first.